### PR TITLE
feat(stdlib): support microseconds in from_unix_timestamp

### LINKED ISSUE #491: Support microseconds in from_unix_timestamp
Hi,

As of version 0.4.31, chrono has a function that takes a UNIX timestamps in microseconds and returns a DateTime struct: [Utc.timestamp_micros](https://docs.rs/chrono/latest/chrono/offset/trait.TimeZone.html#method.timestamp_micros).
I want to add support for using UNIX microsecond timestamps in `from_unix_timestamp.rs`. I forked the repository and implemented my request [here](https://github.com/vectordotdev/vrl/compare/main...final-israel:vrl:support_microseconds_in_from_unix_timestamp).

With this, would vector users be able to use the new `Unit::Microseconds` in their vrl remaps?
Is my implementation acceptable?

### PR DESCRIPTION
**Pull request by emikitas**
resolves #491

### CODE REVIEW

File: src/stdlib/from_unix_timestamp.rs [hunk 1]
```diff
@@ -198,6 +205,12 @@ mod tests {
             tdef: TypeDef::timestamp().fallible(),
         }
 
+        integer_microseconds {
```
Comment (pront) [inline src/stdlib/from_unix_timestamp.rs:208]: Let's also add an example (line 74).

**Review by pront — APPROVED** [review]
Thank you for this contribution @emikitas. This looks good.

**Comment by emikitas** [comment]
I noticed I botched one of my commit messages, so I decided to squash the commit adding the example into it and amend the commit message.

**Comment by pront** [comment]
> I noticed I botched one of my commit messages, so I decided to squash the commit adding the example into it and amend the commit message.

Hi @emikitas, feel free to push as many commits as you need. In the end will squash all those. However, make sure to follow https://www.conventionalcommits.org/en/v1.0.0/ for the final commit.

**Pull request merged**